### DISCUSSION TITLE
Same day pickup constraint

### DISCUSSION THREAD

**Discussion by jaswanthvishnu10** [google/or-tools#2791] [Routing (and legacy CP) questions]
Hi guys,
I am trying to solve a VRP where the vehicles are allowed to pickup at multiple locations and drop at a destination. I am trying to add a constraint such that a vehicle can only perform same day pickup.
I have orders that can be picked up over a span of days. 

Ex: 
Order1 - 1st Oct - 4th Oct,
Order2 - 3rd Oct - 7th Oct

If a vehicle starts pickup on day X, it should only be able to pickup orders that are available for pickup on that day X.
Ex: if Vehicle1 - starts pickup on 3rd October, it can pickup Order1 and Order2 on 3rd October only.

Right now the model is able to do multi pickup and drop. But it is picking up across days.
Can anyone help me to enforce this same day pickup constraint?

**Comment by Mizux** [comment]
you can try to limit the span of the vehicle on one day.

How did you code time ?

**Reply by jaswanthvishnu10** [reply]
I have used time as time in minutes.
[2021-10-01T12:00:00+00:00, 2021-10-02T18:00:00+00:00] -> [27218160, 27219960].

The vehicle can travel for more than a day from a pickup to a drop location, but the pickups should all happen on the same day.

**Reply by jaswanthvishnu10** [reply]
| Pickup node  | Pickup from | Pickup To|
| ------------- | ------------- |-------|
| 1  | 1st Oct  | 4th Oct |
| 2  | 3rd Oct  | 7th Oct |

Destination could be 3 days away from pickup node. 

I am expecting a solution with pickup at node1 and node2 on 3rd October and deliver at destination on 6th October.

**Reply by Mizux** [reply]
if all pickups are at the start of the route you could had a constraint that all pickup of the vehicle X must be performed in the range `[time(routing.Start(X)), time(routing.Start(X))+ 1440]` ?

**Reply by jaswanthvishnu10** [reply]
Yes, that's what i am trying to achieve. 
All pickups happen at the start of the route.
VehicleStart -> Pickup1 -> Pickup2 -> Pickup3 -> Drop1 -> Drop2 -> Drop3 -> VehicleEnd
